components/smalltalk/squeak5/Makefile

```diff
@@ -0,0 +1,95 @@
+#
+
+#
+# Package Makefile for the "OpenSmalltalk Cog Spur" Squeak Smalltalk V5 system
+# See http://squeak.org and http://opensmalltalk.org
+#
+
+BUILD_BITS=64
+COMPILER=gcc
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		squeak5
+COMPONENT_VERSION=	5.0.6
+COMPONENT_SUMMARY=	The Squeak V5 Smalltalk System
+COMPONENT_PROJECT_URL=	http://www.squeak.org
+COMPONENT_FMRI=		runtime/squeak5
+COMPONENT_CLASSIFICATION=	Development/Other Languages
+COMPONENT_LICENSE=	LICENSE
+COMPONENT_SRC=		opensmalltalk-vm
+
+GIT=			git
+GIT_REPO=		https://github.com/cstes/opensmalltalk-vm.git
+GIT_BRANCH=		sun
+GIT_TAG=		sun-v5.0.6
+
+TEST_TARGET= $(NO_TESTS)
+
+include $(WS_MAKE_RULES)/common.mk
+
+PATH=$(PATH.gnu)
+
+# The ugly hack with update-publish target is necessary to update
+# source from git repository on each "gmake publish".
+publish: update-publish
+
+# the opensmalltalk README says to run scripts/updateSCCSVersions 
+# but I believe that we can't do this on unversioned git files
+# so we must build from git sources, I believe
+
+$(SOURCE_DIR)/.downloaded: $(ARCHIVES:%=$(USERLAND_ARCHIVES)%)
+	@[ -d $(SOURCE_DIR) ] || \
+	$(GIT) clone -b $(GIT_BRANCH) $(GIT_REPO) $(SOURCE_DIR)
+	@cd $(SOURCE_DIR); $(GIT) checkout $(GIT_TAG); \
+	$(GIT) log -1 --format=%H > .downloaded
+
+# I don't understand the hack but I manually verify the download
+update-publish::
+
+download:: $(SOURCE_DIR)/.downloaded
+
+# opensmalltalk configure script checks for plugins files in builddir
+COMPONENT_PRE_CONFIGURE_ACTION = ( cp plugins.ext plugins.int $(BUILD_DIR_64); cd $(SOURCE_DIR); ./scripts/updateSCCSVersions  )
+
+# the default CFLAGS.gcc will be -m64 -O3 but this does not work for us
+# opensmalltalk code is not compatible with the gcc optimizer -O2 or higher
+# also the -g (debug) flag is intentional, perhaps omit-frame-pointer issues
+CFLAGS=			-g -DNDEBUG -DDEBUGVM=0
+
+CONFIGURE_SCRIPT= $(SOURCE_DIR)/platforms/unix/config/configure
+CONFIGURE_OPTIONS +=	--with-vmversion=5.0
+CONFIGURE_OPTIONS +=	--disable-dynamicopenssl
+CONFIGURE_OPTIONS +=	--with-src=spurstack64src
+CONFIGURE_OPTIONS +=	--disable-cogit
+CONFIGURE_OPTIONS +=	--without-vm-display-fbdev
+CONFIGURE_OPTIONS +=	--without-npsqueak
+
+CONFIGURE_ENV +=	TARGET_ARCH="-m64"
+
+# the opensmalltalk Makefile does not use DESTDIR
+COMPONENT_INSTALL_ARGS=  	ROOT=$(PROTO_DIR)
+COMPONENT_INSTALL_TARGETS= 	install-squeak install-doc install-plugins
+
+REQUIRED_PACKAGES += service/opengl/ogl-select
+REQUIRED_PACKAGES += system/library/dbus
+
+# Auto-generated dependencies
+REQUIRED_PACKAGES += SUNWcs
+REQUIRED_PACKAGES += library/audio/pulseaudio
+REQUIRED_PACKAGES += library/desktop/cairo
+REQUIRED_PACKAGES += library/desktop/pango
+REQUIRED_PACKAGES += library/glib2
+REQUIRED_PACKAGES += library/security/openssl
+REQUIRED_PACKAGES += system/library
+REQUIRED_PACKAGES += system/library/math
+REQUIRED_PACKAGES += x11/library/libx11
+REQUIRED_PACKAGES += x11/library/libxrender
```

Comment:
Something seems to be wrong or missing: "gmake build" fails on my build machine:
libtool: link: ( cd ".libs" && rm -f "UUIDPlugin.la" && ln -s "../UUIDPlugin.la" "UUIDPlugin.la" )
: XDisplayControlPlugin.la
libtool: link: /usr/gcc/7/bin/gcc -shared  -fPIC -DPIC -Wl,-z -Wl,text -Wl,-h -Wl,SqueakSSL.so -o .libs/SqueakSSL.so  .libs/SqueakSSL.o .libs/sqUnixSSL.o   -L/usr/local/lib -lm -lsocket -lnsl -lcrypto -lssl -ltls  -g -pthread -m64 -m64 -Wl,-z -Wl,now   -pthread
: UUIDPlugin.la
/oibuild/oi-userland/components/smalltalk/squeak5/opensmalltalk-vm/platforms/unix/config/verstamp version.c /usr/gcc/7/bin/gcc
ld: fatal: library -ltls: not found
ld: fatal: file processing errors. No output written to .libs/SqueakSSL.so
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:169: SqueakSSL.la] Error 1
make[1]: *** [Makefile:541: SqueakSSL.la] Error 2

Reply:
Looks like a configure issue where configure is perhaps enabling --with-tls on your server.

I did not on my system.

From looking at the configure script it is checking for a header file tls.h.

Do you :  find /usr/include/openssl -name tls.h on your system ?

I'll check tomorrow but it built for me and best guess is that somehow the openssl configure logic of "opensmalltalk" is enabling the tls support (by mistake) either by detecting the header file or due to some other reason ...

Reply:
What is the result on your system of

pkg search -af tls.h

??   is there perhaps a package installed that offers a tls.h ?

My guess is that the opensmalltalk-vm  configure script is confused and enables --with-tls in your case.

To avoid this (if it is the problem) the Makefile would have to "exclude" some packages that may not be installed I guess.

Reply:
One possibility is perhaps

./tcltls/tcltls.p5m:file path=usr/include/tls.h

tcltls is it installed for you ?  anyway I'll check tomorrow

also depending on your point of view this can be considered a problem with the opensmalltalk-vm configure script  -- or a problem with the build server - or the Makefile of squeak5.p5m that could perhaps (if possible) exclude packages that may not be installed

Reply:
Possibly adding --without-libtls  helps ... 
I've just made that modification to the Makefile but I will have to test whether it then still builds for me.   For me it built without that option.

Reply:
I have almost any package installed on my build system.
─➤  pkg search -l   /usr/include/tls.h
INDEX      ACTION VALUE             PACKAGE
path       file   usr/include/tls.h pkg:/runtime/tcl-8/tcl-openssl@1.6.7-2020.0.1.0

--without-libtls fixed the problem.